pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/GetPartitionMetadataTest.java

```diff
@@ -0,0 +1,289 @@
+        conf.setAllowAutoTopicCreationType(TopicType.PARTITIONED);
+        conf.setDefaultNumPartitions(3);
+        conf.setAllowAutoTopicCreation(true);
+        setup();
+        LookupService lookup = getLookupService(isUsingHttpLookup);
+        // Create topic.
+        final String topicNameStr = BrokerTestUtil.newUniqueName("persistent://" + DEFAULT_NS + "/tp");
+        final TopicName topicName = TopicName.get(topicNameStr);
+        // Verify.
+        PulsarClient client = PulsarClient.builder().serviceUrl(pulsar.getBrokerServiceUrl()).build();
+        PartitionedTopicMetadata response = lookup.getPartitionedTopicMetadata(topicName, true).join();
+        assertEquals(response.partitions, 3);
+        List<String> partitionedTopics = admin.topics().getPartitionedTopicList("public/default");
+        assertTrue(partitionedTopics.contains(topicNameStr));
+        List<String> topicList = admin.topics().getList("public/default");
+        assertFalse(topicList.contains(topicNameStr));
+        for (int i = 0; i < 3; i++) {
+            // The API "getPartitionedTopicMetadata" only creates the partitioned metadata, it will not create the
+            // partitions.
+            assertFalse(topicList.contains(topicName.getPartition(i)));
+        }
+        // Cleanup.
+        client.close();
+        admin.topics().deletePartitionedTopic(topicNameStr, false);
+    }
+
+    @Test(dataProvider = "clients")
+    public void testAutoCreateNonPartitionedTopic(boolean isUsingHttpLookup) throws Exception {
+        conf.setAllowAutoTopicCreationType(TopicType.NON_PARTITIONED);
+        conf.setAllowAutoTopicCreation(true);
+        setup();
+        LookupService lookup = getLookupService(isUsingHttpLookup);
+        // Create topic.
+        final String topicNameStr = BrokerTestUtil.newUniqueName("persistent://" + DEFAULT_NS + "/tp");
+        final TopicName topicName = TopicName.get(topicNameStr);
+        // Verify.
+        PulsarClient client = PulsarClient.builder().serviceUrl(pulsar.getBrokerServiceUrl()).build();
+        PartitionedTopicMetadata response = lookup.getPartitionedTopicMetadata(topicName, true).join();
+        assertEquals(response.partitions, 0);
+        List<String> partitionedTopics = admin.topics().getPartitionedTopicList("public/default");
+        assertFalse(partitionedTopics.contains(topicNameStr));
+        List<String> topicList = admin.topics().getList("public/default");
+        assertFalse(topicList.contains(topicNameStr));
+        // Cleanup.
+        client.close();
+    }
+
+    @DataProvider(name = "autoCreationParamsNotAllow")
+    public Object[][] autoCreationParamsNotAllow(){
+        return new Object[][]{
+                // configAllowAutoTopicCreation, paramCreateIfAutoCreationEnabled, isUsingHttpLookup.
+                {true, false, true},
+                {true, false, false},
+                // {false, true, true},
+                // {false, true, false},
```

Comment:
Any reason for adding `//` for these 2 line?

Reply:
Oh, sorry. I tried to debug the test, `//` should be removed. I removed `//`

Reply:
> These test cases are for the following PR, which was described in the https://github.com/apache/pulsar/pull/22206#issue-2170535985, I implemented it in this PR now.

Since some functions rely on the original behavior and lead many tests to fail, I rollbacked this change. The change will be in the next PR.